src/apps/airswap/ethereum/airswap.s-ast.token-fetcher.ts

```diff
@@ -0,0 +1,34 @@
+import { Inject } from '@nestjs/common';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { AppTokenPosition, ContractPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+
+import { AirswapContractFactory } from '../contracts';
+import { AIRSWAP_DEFINITION } from '../airswap.definition';
+
+const appId = AIRSWAP_DEFINITION.id;
+const groupId = AIRSWAP_DEFINITION.groups.sAST.id;
+const network = Network.ETHEREUM_MAINNET;
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class EthereumAirswapSAstTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(AirswapContractFactory) private readonly airswapContractFactory: AirswapContractFactory,
+  ) {}
+
+  async getPositions() {
+    const address = '0x579120871266ccd8De6c85EF59E2fF6743E7CD15';
+
+    // How to get user account?
+    const account = '0xfd72d3071e1056aad6d076518c483200d2f9ad65';
+
+    const contract = this.airswapContractFactory.staking({ address, network });
+    const balance = await contract.balanceOf(account);
+
+    return [balance];
```

Comment:
Not sure what to do here. How to properly return the balance?

Reply:
Hey @piersss !

The `EthereumAirswapSAstTokenFetcher ` class should simply be returning an `AppTokenPosition` that represents the `sAST` token. You can follow along [here](https://studio-docs.zapper.fi/docs/tutorial/create-token-fetcher) for details.

Once you read that, you'll see that:
- You'll need to build your `AppTokenPosition` for `sAST` with `AST` as an underlying token
- Your `pricePerShare` would be `1` since users mint `sAST` in a 1:1 ratio with deposited `AST`

Then, you can update your `EthereumAirswapBalanceFetcher` class following along [here](https://studio-docs.zapper.fi/docs/tutorial/create-balance-fetcher). The `BalanceFetcher` receives the _user account_ as you mentioned above.

Reply:
Thank you @immasandwich . Making progress here.

Everything looks fine in `http://localhost:5001/apps/airswap/balances?addresses[]=<ADDRESS>&network=ethereum`. Is there a way to test the dashboard and see if sAST shows up?

Reply:
Not yet unfortunately! But we're quite confident in it working correctly if you're correct on the types being returned.

Looks good to me! Just a few more things:
- Add the Airswap logo (the AST token image) in `airswap/assets/logo.png`
- Delete the `yarn.lock` file you generated (we use `pnpm-lock.json` instead)

Reply:
Once that's good, we'll merge and do a Studio release, and we'll consume the latest version in Zapper API.

Reply:
Fixed your feedback, thanks.

The types returned looks correct. It returns the sAST with AST as underlying token.